gr-soapy/python/soapy/bindings/soapy_common.cc

```diff
@@ -49,22 +51,20 @@ static inline T string_to_setting(const std::string& str)
 }
 
 // Copied from SoapySDR 0.8
+//! convert empty and "false" strings to false, integers to their truthness
 template <>
 inline bool string_to_setting<bool>(const std::string& str)
 {
-    if (str == SOAPY_SDR_TRUE)
-        return true;
-    if (str == SOAPY_SDR_FALSE)
-        return false;
-
-    // zeros and empty strings are false
-    if (str == "0")
-        return false;
-    if (str == "0.0")
-        return false;
-    if (str == "")
+    if (str.empty() || str == SOAPY_SDR_FALSE) {
         return false;
-
+    }
+    if (str == SOAPY_SDR_TRUE) {
+        return true;
+    }
+    try {
+        return std::stod(str);
```

Comment:
Even though it comes out the same, can you please:
```suggestion
        return static_cast<bool>(std::stod(str));
```
MSVC warns about this otherwise, but I also think it tells a better story when reading the code.

Reply:
very true